src/sonic-config-engine/portconfig.py

```diff
@@ -250,6 +250,14 @@ def parse_platform_json_file(hwsku_json_file, platform_json_file):
         brkout_mode = hwsku_dict[INTF_KEY][intf][BRKOUT_MODE]
 
         child_ports = get_child_ports(intf, brkout_mode, platform_json_file)
+
+        # take optional fields from hwsku.json
+        for key, item in hwsku_dict[INTF_KEY][intf].items():
```

Comment:
I think it is probably better to take conservative way to introduce the new fields in the hwsku.json by explicitly define the new fields in a structure, and only allow the defined fields to be generated to PORT table.   To make the detection earlier, we can also add something like https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-device-data/tests/platform_json_checker to do sanity check for hwsku.json at build time.